metadata-ingestion/src/datahub/utilities/bigquery_sql_parser.py

```diff
@@ -0,0 +1,69 @@
+from typing import List
+import re
+import sqlparse
+from datahub.utilities.sql_parser import SQLParser, SqlLineageSQLParser
+
+class BigQuerySQLParser(SQLParser):
+    parser: SQLParser
+
+    def __init__(self, sql_query: str) -> None:
+        super().__init__(sql_query)
+
+        self._parsed_sql_query = self.parse_sql_query(sql_query)
+        self.parser = SqlLineageSQLParser(self._parsed_sql_query)
+
+    def parse_sql_query(self, sql_query) -> str:
+        sql_query = BigQuerySQLParser._parse_bigquery_comment_sign(sql_query)
+        sql_query = BigQuerySQLParser._escape_keyword_from_as_field_name(sql_query)
+        sql_query = BigQuerySQLParser._escape_cte_name_after_keyword_with(sql_query)
+
+        sql_query = sqlparse.format(
+            sql_query.strip(),
+            reindent_aligned=True,
+            strip_comments=True,
+        )
+
+        sql_query = BigQuerySQLParser._escape_table_or_view_name_at_create_statement(sql_query)
+        sql_query = BigQuerySQLParser._escape_object_name_after_keyword_from(sql_query)
+
+        return sql_query
+
+    @staticmethod
+    def _parse_bigquery_comment_sign(sql_query) -> str:
+        return re.sub(r"#(.*)", r"-- \1", sql_query, flags=re.IGNORECASE)
+
+    @staticmethod
+    def _escape_keyword_from_as_field_name(sql_query) -> str:
+        return re.sub(r"(\w*\.from)", r"`\1`", sql_query, flags=re.IGNORECASE)
+
+    @staticmethod
+    # Escape the first cte name in case it is one of reserved words
```

Comment:
Comments describing functions are written in triple quotes in Python. Please see `bigquery.py` for examples. You made changes in one such comment.

Reply:
Done